### ISSUE TITLE
IPA (International Phonetic Alphabet) voice recognition and text to speech

### ISSUE THREAD

**Issue by Kreijstal** [open-source-ideas/open-source-ideas#134]
## Project description

[_Describe the project the best you can. Give any background information or link to resources that are necessary to understand the problem it is intended to solve. The more you elaborate on your idea, the easier it is to accomplish._]

This is essentially the same as voice recognition except easier because programs will only listen for pronunciation without attempting to map it to a word.

## Relevant Technology

[_Write what technology is relevant. What language, what platform, any particular library/framework/existing project it is based on?_]

Probably the same techniques used in traditional voice recognition

## Complexity and required time

[_Please only tick off one box in each category by changing `[ ]` to `[x]`. The labels on the project will then be updated by the maintainers as soon as possible._]

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [ ] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [x] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ ] Medium work - A week or two
- [x] Much work - The project will take more than a couple of weeks and serious planning is required

**Comment by straight-shoota** [comment]
> easier because programs will only listen for pronunciation without attempting to map it to a word.

That's hardly true. Speech recognition is easier when mapped to a specific language because knowing the linguistic features of a language plus the vocabulary of that language helps a lot with recognizing otherwise ambiguous phrases.

There are alone 34 different vowels sounds which would need to be differentiated. Making voice recognition understand the fine nuances in the phonetic alphabet without any guidelines of a language framework is most likely not solvable anytime in the near future.

**Comment by dshiryu** [comment]
> > easier because programs will only listen for pronunciation without attempting to map it to a word.
> 
> That's hardly true. Speech recognition is easier when mapped to a specific language because knowing the linguistic features of a language plus the vocabulary of that language helps a lot with recognizing otherwise ambiguous phrases.
> 
> There are alone 34 different vowels sounds which would need to be differentiated. Making voice recognition understand the fine nuances in the phonetic alphabet without any guidelines of a language framework is most likely not solvable anytime in the near future.

It really depends on what the intended use is. While the project name says "text to speech", considering the IPA observation, one might consider the end goal isn't an Alexa of sorts, but instead one specific for linguistic studies.

In a phonetics class back in my linguistics degree we had to transcribe the IPA sounds from interviews (not the words said, the sounds used according to the IPA table), so converting the sounds in transcription automatically could save researchers many hours of work and potentially help the field advance faster (e.g. it took me at least 2 hours to transcribe phonetically a 5-minute interview).  My university had a corpus of interviews a couple decades old that wasn't nearly started being transcripted, for instance, due to lack of resources to invest in it.

Tbh I've started a major in CS and have something like I just described as an idea of how to mix both studies. I found this page while making sure the idea wasn't taken already. It seems I'm not too late though lol

**Comment by Kreijstal** [comment]
> > easier because programs will only listen for pronunciation without attempting to map it to a word.
> 
> That's hardly true. Speech recognition is easier when mapped to a specific language because knowing the linguistic features of a language plus the vocabulary of that language helps a lot with recognizing otherwise ambiguous phrases.
> 
> There are alone 34 different vowels sounds which would need to be differentiated. Making voice recognition understand the fine nuances in the phonetic alphabet without any guidelines of a language framework is most likely not solvable anytime in the near future.

I guess the idea is to make a 1 to 1 map of  audio sounds you can make to symbols.

**Comment by bastienboutonnet** [comment]
Oooeh! Might be a fun thing to try during Christmas holiday! Do you have a use case in mind @Kreijstal ?

**Comment by dshiryu** [comment]
> Oooeh! Might be a fun thing to try during Christmas holiday! Do you have a use case in mind @Kreijstal ?

Considering what he mentioned about the 1 to 1 map, since I have a (unused) Linguistics degree, I'd guess one possibility is taking a given audio recording and giving back the transcription. For instance: 
"The runner crossed the finishing line." returns /ðə ˈrʌnər krɔst ðə ˈfɪnɪʃɪŋˈlaɪn/ (probably without spaces without a dictionaty implemented).

I'm guessing that the "easiest" way to tackle this is with AI, right? Basically take the sounds from https://www.ipachart.com/ as a based to know which symbols each sound uses, then add the training corpus and let it finish its magic (however that works, I'm still working on my CS degree lol)

**Comment by Kreijstal** [comment]
> Oooeh! Might be a fun thing to try during Christmas holiday! Do you have a use case in mind @Kreijstal ?

Yes, basically, a language agnostic way of transcribing human made sounds. Regardless of language, you could then with the written symbols try to map what they say (in symbolic) form to how it is written.  Or you could do the inverse and have a TTS that can  pronounce everything out of the box, in any language, because it doesn't speak a language, it just speaks sounds. Even weird ones, or sounds that don't belong to any language.

**Comment by KOLANICH** [comment]
Text-to-Speech Synthesis Paul Taylor 978-0-521-89927-7 is highly relevant.
